pkg/agent/externalnode/external_node_controller_linux_test.go

```diff
@@ -0,0 +1,1155 @@
+			}
+			addInterfaceTimes := 0
+			if tt.ifAddInterface {
+				addInterfaceTimes = 1
+			}
+			mockIfaceStore.EXPECT().AddInterface(iface).Times(addInterfaceTimes)
+			if !tt.ifaceExist {
+				defer mockGetInterfaceConfig([]mockGetInterfaceConfigParam{tt.getInterfaceConfigParam})()
+				defer mockConfigureLinkRoutes(nil)()
+				defer mockConfigureLinkAddresses(nil)()
+				defer mockRemoveLinkRoutes(nil)()
+				defer mockRenameInterface(nil)()
+				defer mockRemoveLinkIPs(nil)()
+				defer mockLinkSetUp(nil)()
+				defer mockLinkSetMTU(nil)()
+				defer mockLinkByName(t, 2)()
+				uplinkName := tt.iface.InterfaceName + "~"
+				mockOVSBridgeClient.EXPECT().CreatePort(
+					uplinkName,
+					uplinkName,
+					map[string]interface{}{interfacestore.AntreaInterfaceTypeKey: interfacestore.AntreaUplink},
+				).Return(tt.iface.UplinkPort.PortUUID, nil).Times(1)
+				expectedInfo := map[string]interface{}{
+					"uplink-name":      uplinkName,
+					"entity-name":      tt.eeName,
+					"antrea-type":      "host",
+					"entity-namespace": tt.eeNamespace,
+					"ip-address":       strings.Join(tt.preIPs, ipsSplitter),
+					"uplink-port":      tt.iface.UplinkPort.PortUUID,
+				}
+				mockOVSBridgeClient.EXPECT().CreateInternalPort(tt.iface.InterfaceName, int32(0), tt.getInterfaceConfigParam.iface.HardwareAddr.String(), expectedInfo).Return(tt.iface.PortUUID, nil)
+				mockOVSBridgeClient.EXPECT().GetOFPort(tt.iface.InterfaceName, false).Times(1).Return(tt.iface.OFPort, nil)
+				mockOVSBridgeClient.EXPECT().GetOFPort(uplinkName, false).Times(1).Return(tt.iface.UplinkPort.OFPort, nil)
+				mockOFClient.EXPECT().InstallVMUplinkFlows(tt.iface.InterfaceName, tt.iface.OFPort, tt.iface.UplinkPort.OFPort).Times(1)
+				c.addInterface(tt.iface.InterfaceName, tt.eeNamespace, tt.eeName, tt.preIPs)
+				return
+			}
+			portData := &ovsconfig.OVSPortData{
+				ExternalIDs: map[string]string{
+					ovsExternalIDUplinkName:      tt.uplinkName,
+					ovsExternalIDUplinkPort:      tt.uplinkPortUUID,
+					ovsExternalIDEntityName:      tt.preEEName,
+					ovsExternalIDEntityNamespace: tt.eeNamespace,
+					ovsExternalIDIPs:             strings.Join(tt.preIPs, ipsSplitter),
+				},
+			}
+			mockOVSBridgeClient.EXPECT().GetPortData(tt.iface.PortUUID, tt.iface.InterfaceName).Return(portData, nil).Times(1)
+			expectedInfo := map[string]interface{}{
+				"uplink-name":      tt.uplinkName,
+				"entity-name":      tt.eeName,
+				"antrea-type":      "host",
+				"entity-namespace": tt.eeNamespace,
+				"ip-address":       strings.Join(tt.IPs, ipsSplitter),
+				"uplink-port":      tt.uplinkPortUUID,
+			}
+			mockOVSBridgeClient.EXPECT().SetPortExternalIDs(if1.InterfaceName, expectedInfo)
+			c.addInterface(tt.iface.InterfaceName, tt.eeNamespace, tt.eeName, tt.IPs)
+		})
+	}
+}
+
+func TestCreateOVSPortsAndFlowsFailure(t *testing.T) {
+	c := newExternalNodeController(t)
+
+	_, addr1, _ := net.ParseCIDR("10.20.30.40")
```

Comment:
Better to name it `cidr1`, and probably move it to `var` section since you defined this in next test function as well.